### PR TITLE
GUI fixes + log stacking

### PR DESCRIPTION
**Pull request by Megamouse**
@AniLeo @flash-fire please review

log:
- adds stacking option for same log messages to log contextmenu.
- fix log color change for issue #2851
- fix scrolling and selection

menu:
- fixes a bug that was preventing the "show menu" to show the right state
- fix "recent games" crash

gamelist:
- stretch to full range
- add sort indicator
- add background color to icons for better readability
- add default layout for better looks on first start
- remove highlight on selection aka bold headers
- add 3 more columns

### CODE REVIEW

File: rpcs3/Emu/System.h [hunk 1]
```diff
@@ -377,6 +377,7 @@ struct cfg_root : cfg::node
 		cfg::_bool autoexit{this, "Exit RPCS3 when process finishes"};
 		cfg::_bool start_fullscreen{ this, "Start games in fullscreen mode" };
 		cfg::_bool show_fps_in_title{ this, "Show FPS counter in window title", true};
+		cfg::_bool stack_log_messages{ this, "Stack messages in log frame", false };
```
Comment (Nekotekina) [inline rpcs3/Emu/System.h]: This has nothing to do with RPCS3 core config!

**Comment by flash-fire** [comment]
Looks fine assuming that this gets squashed or rebased before merge.

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 2]
```diff
@@ -910,3 +934,33 @@ std::string game_list_frame::CurrentSelectionIconPath()
 
 	return selection;
 }
+
+std::string game_list_frame::GetStringFromU32(const u32& key, const std::map<u32, QString> map, bool combined)
```
Comment (danilaml) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: >const std::map<u32, QString> map

missing `&`

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 3]
```diff
@@ -910,3 +934,33 @@ std::string game_list_frame::CurrentSelectionIconPath()
...
+{
+	QStringList string;
+
+	if (combined)
+	{
+		for (auto item : map)
```
Comment (danilaml) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: `const auto &`

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 4]
```diff
@@ -354,10 +372,10 @@ void game_list_frame::Refresh(bool fromDrive)
 			GameInfo game;
 			game.root         = entry.name;
 			game.serial       = psf::get_string(psf, "TITLE_ID", "");
-			game.name         = psf::get_string(psf, "TITLE", "unknown");
-			game.app_ver      = psf::get_string(psf, "APP_VER", "unknown");
-			game.category     = psf::get_string(psf, "CATEGORY", "unknown");
-			game.fw           = psf::get_string(psf, "PS3_SYSTEM_VER", "unknown");
+			game.name         = psf::get_string(psf, "TITLE", sstr(category::unknown));
```
Comment (danilaml) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: `sstr` function is totally unnecessary. Just use `toStdString()`
http://doc.qt.io/qt-5/qstring.html#toStdString
If you want to keep the shorted conversion function, then `toUtf8` part is unnecessary according to: http://doc.qt.io/qt-5/qstring.html#toStdString

Comment (Megamouse) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: since we do this in many other places as well, we will fix this seperately in another commit if necessary

**Pull request merged**